packages/data/src/plugins/actions.js

```diff
@@ -125,6 +126,52 @@ export function* installAndActivatePlugins( plugins ) {
 	}
 }
 
+export const createErrorNotice = ( errorMessage ) => {
+	return dispatch( 'core/notices', 'createNotice', errorMessage );
+};
+
+export function* connectToJetpack() {
+	const url = yield select( STORE_NAME, 'getJetpackConnectUrl', {
+		redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+	} );
+	const error = yield select(
+		STORE_NAME,
+		'getPluginsError',
+		'getJetpackConnectUrl'
+	);
+
+	if ( error ) {
+		throw new Error( error );
+	} else {
+		return url;
+	}
+}
+
+export function* installJetpackAndConnect( errorAction ) {
+	try {
+		yield dispatch( STORE_NAME, 'installPlugins', [ 'jetpack' ] );
+		yield dispatch( STORE_NAME, 'activatePlugins', [ 'jetpack' ] );
+
+		const url = yield dispatch( STORE_NAME, 'connectToJetpack' );
+		window.location = url;
+	} catch ( error ) {
+		yield errorAction( error.message );
```

Comment:
Nice!  I really like the simplicity of the callback error here.  I think ultimately moving towards throwing errors in all actions to make them consistently and catchable would be ideal, but we have an issue logged for this and can follow-up later.